From a8c7447fe2078c28d0c3494fc5659012eba9367e Mon Sep 17 00:00:00 2001
From: Mitul Chokshi <mitul.chokshi@intel.com>
Date: Thu, 30 Mar 2017 17:01:57 +0100
Subject: [PATCH 0454/1240] drm/i915/gvt: Added memory latency value for BXT
 platform

Value is acquired from read memory latency command on BXT MRB

Signed-off-by: Mitul Chokshi <mitul.chokshi@intel.com>
Reviewed-by: Singh, Satyeshwar <satyeshwar.singh@intel.com>
(cherry picked from commit 437e34f49b0a082e8bd421ebcefa8d335e6b4863)

Change-Id: Ibe2cb6203e4b094f3dd660d94792d03c478ee8b6
Reviewed-by: He, Min <min.he@intel.com>
Reviewed-by: Jiang, Fei <fei.jiang@intel.com>
Reviewed-by: Dong, Eddie <eddie.dong@intel.com>
Tested-by: Dong, Eddie <eddie.dong@intel.com>
---
 drivers/gpu/drm/i915/gvt/handlers.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/gpu/drm/i915/gvt/handlers.c b/drivers/gpu/drm/i915/gvt/handlers.c
index 8c51cf0..708a144 100755
--- a/drivers/gpu/drm/i915/gvt/handlers.c
+++ b/drivers/gpu/drm/i915/gvt/handlers.c
@@ -1379,6 +1379,16 @@ static int mailbox_write(struct intel_vgpu *vgpu, unsigned int offset,
 				*data0 = 0x1e1a1100;
 			else
 				*data0 = 0x61514b3d;
+		} else if(IS_BROXTON(vgpu->gvt->dev_priv)) {
+			/**
+			 * "Read memory latency" command on gen9.
+			 * Below memory latency values are read
+			 * from broxton MRB.
+			 */
+			if (!*data0)
+				*data0 = 0x16080707;
+			else
+				*data0 = 0x16161616;
 		}
 		break;
 	case SKL_PCODE_CDCLK_CONTROL:
-- 
2.7.4

